lib/block_view.go

```diff
@@ -7063,89 +7408,96 @@ func (bav *UtxoView) _connectAcceptNFTBid(
+			return 0, 0, nil, errors.Wrapf(RuleErrorCantAcceptNonExistentBid, "_helpConnectNFTSold: ")
 		}
 	}
 	if nftBidEntry.BidAmountNanos != txMeta.BidAmountNanos {
-		return 0, 0, nil, RuleErrorAcceptedNFTBidAmountDoesNotMatch
+		return 0, 0, nil, errors.Wrapf(RuleErrorAcceptedNFTBidAmountDoesNotMatch, "_helpConnectNFTSold: ")
 	}
 
 	bidderPublicKey := bav.GetPublicKeyForPKID(txMeta.BidderPKID)
-
 	//
 	// Store starting balances of all the participants to check diff later.
 	//
 	// We assume the tip is right before the block in which this txn is about to be applied.
 	tipHeight := uint32(0)
+	blockHeight := txMeta.BlockHeight
 	if blockHeight > 0 {
 		tipHeight = blockHeight - 1
 	}
-	sellerBalanceBefore, err := bav.GetSpendableDeSoBalanceNanosForPublicKey(txn.PublicKey, tipHeight)
+	sellerPublicKey := bav.GetPublicKeyForPKID(prevNFTEntry.OwnerPKID)
+	sellerBalanceBefore, err := bav.GetSpendableDeSoBalanceNanosForPublicKey(sellerPublicKey, tipHeight)
 	if err != nil {
 		return 0, 0, nil, fmt.Errorf(
-			"_connectAcceptNFTBid: Problem getting initial balance for seller pubkey: %v",
-			PkToStringBoth(txn.PublicKey))
+			"_helpConnectNFTSold: Problem getting initial balance for seller pubkey: %v",
+			PkToStringBoth(sellerPublicKey))
 	}
 	bidderBalanceBefore, err := bav.GetSpendableDeSoBalanceNanosForPublicKey(
 		bidderPublicKey, tipHeight)
 	if err != nil {
 		return 0, 0, nil, fmt.Errorf(
-			"_connectAcceptNFTBid: Problem getting initial balance for bidder pubkey: %v",
+			"_helpConnectNFTSold: Problem getting initial balance for bidder pubkey: %v",
 			PkToStringBoth(bidderPublicKey))
 	}
 	creatorBalanceBefore, err := bav.GetSpendableDeSoBalanceNanosForPublicKey(
 		nftPostEntry.PosterPublicKey, tipHeight)
 	if err != nil {
 		return 0, 0, nil, fmt.Errorf(
-			"_connectAcceptNFTBid: Problem getting initial balance for poster pubkey: %v",
+			"_helpConnectNFTSold: Problem getting initial balance for poster pubkey: %v",
 			PkToStringBoth(nftPostEntry.PosterPublicKey))
 	}
-
-	//
-	// Validate bidder UTXOs.
-	//
-	if len(txMeta.BidderInputs) == 0 {
-		return 0, 0, nil, RuleErrorAcceptedNFTBidMustSpecifyBidderInputs
-	}
-	totalBidderInput := uint64(0)
+	bidderChangeNanos := uint64(0)
 	spentUtxoEntries := []*UtxoEntry{}
 	utxoOpsForTxn := []*UtxoOperation{}
-	for _, bidderInput := range txMeta.BidderInputs {
-		bidderUtxoKey := UtxoKey(*bidderInput)
-		bidderUtxoEntry := bav.GetUtxoEntryForUtxoKey(&bidderUtxoKey)
-		if bidderUtxoEntry == nil || bidderUtxoEntry.isSpent {
-			return 0, 0, nil, RuleErrorBidderInputForAcceptedNFTBidNoLongerExists
-		}
+	// We only need to validate the bidder UTXOs when connecting an AcceptNFTBid transaction since the transactor and
+	// the bidder are different users.  For NFTBid transactions on Buy Now NFTs, there are additional inputs to cover
+	// the bid amount. We do not need to make explicitly make change for the bidder in that situation either.
+	if txMeta.IsAcceptNFTBid {
```

Comment:
This would be `if args.TxnType == TxnTypeAcceptNFTBid`

Reply:
yep yep. must smoother. I just fetch it from the Txn instead so that there can't be any trickery.

Reply:
so `Txn.TxnMeta.GetTxnType() == TxnTypeAcceptNFTBid`